### PR TITLE
Avoid SourcePrinter to calculate newline while updating cursor Position

### PR DESCRIPTION
**Pull request by tarilabs**
... by disallow passing strings with newlines to the sourceprinter.

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java [hunk 1]
```diff
@@ -59,68 +52,78 @@ public SourcePrinter unindent() {
...
-
-    public void pushMethodChainPosition(Position position) {
-        this.methodChainPositions.push(position);
-    }
-
-    public Position peekMethodChainPosition() {
-        return this.methodChainPositions.peek();
-    }
-
-    public Position popMethodChainPosition() {
```
Comment (tarilabs) [inline javaparser-core/src/main/java/com/github/javaparser/printer/SourcePrinter.java:120]: All these method are actually just meaningful for PrettyPrinterVisitor, hence moved there.

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java [hunk 2]
```diff
@@ -1348,10 +1367,15 @@ public void visit(final BlockComment n, final Void arg) {
         if (configuration.isIgnoreComments()) {
             return;
         }
-        printer
-                .print("/*")
-                .print(normalizeEolInTextBlock(n.getContent(), configuration.getEndOfLineCharacter()))
-                .println("*/");
+        final String commentContent = normalizeEolInTextBlock(n.getContent(), configuration.getEndOfLineCharacter());
```
Comment (tarilabs) [inline javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java:1370]: > option 3: I think only blockcomment is still printing strings with newlines. What if we disallow passing strings with newlines to the sourceprinter, then do the line splitting in the blockcomment visit method?

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java [hunk 3]
```diff
@@ -1348,10 +1367,15 @@ public void visit(final BlockComment n, final Void arg) {
...
+        String[] lines = commentContent.split("\\R", -1); // as BlockComment should not be formatted, -1 to preserve any trailing empty line if present
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java:1371]: Oh, didn't know about \R. Thorough! :-)

**Comment by matozoid** [comment]
Looks fine to me, and I heard it fixes the performance problems, so here it goes!

File: javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java (hunk 3 above)
Comment (tarilabs) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java:1371]: Borrowed from here https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/utils/Utils.java#L226
;)

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java:1371]: Hehehe, I guess I forgot ;-)

**Pull request merged**